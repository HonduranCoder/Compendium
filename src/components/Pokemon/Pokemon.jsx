//Render Pokemon Item, map through an array in Pokemon list
import React from 'react';

export default function Pokemon({
  pokemon,
  hp,
  attack,
  defense,
  shape,
  image,
}) {
  return (
    <div aria-label="pokemon" key={pokemon}>
      <img src={image} alt="poke" />
      <p>Name: {pokemon}</p>
      <p>HP: {hp}</p>
      <p>Attack: {attack}</p>
      <p>Defense: {defense}</p>
      <p>Shape: {shape}</p>
    </div>
  );
}
