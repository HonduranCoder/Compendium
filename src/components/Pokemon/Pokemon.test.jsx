import { screen, render } from '@testing-library/react';
import Pokemon from './Pokemon.jsx';

it('test sort', () => {
  render(
    <Pokemon pokemon="pikachu" hp={23} attack={40} defense={40} shape="arms" />
  );
  const pokemon = screen.getByLabelText('pokemon');
  expect(pokemon).toMatchSnapshot();
});
