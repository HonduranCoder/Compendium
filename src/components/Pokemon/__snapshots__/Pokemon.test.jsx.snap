// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test sort 1`] = `
<div
  aria-label="pokemon"
>
  <img
    alt="poke"
  />
  <p>
    Name: 
    pikachu
  </p>
  <p>
    HP: 
    23
  </p>
  <p>
    Attack: 
    40
  </p>
  <p>
    Defense: 
    40
  </p>
  <p>
    Shape: 
    arms
  </p>
</div>
`;
